apps/mwiniimporter/importer.cpp

```diff
@@ -910,7 +910,7 @@ void MwIniImporter::importGameFiles(multistrmap &cfg, const multistrmap &ini, co
                     std::time_t time = lastWriteTime(path, defaultTime);
                     if (time != defaultTime)
                     {
-                        contentFiles.push_back({time, path});
+                        contentFiles.emplace_back(time, path);
```

Comment:
```suggestion
                        contentFiles.emplace_back(time, std::move(path));
```
Otherwise `path` is still copied because without explicit casting to rvalue reference it's passed by lvalue reference and copy constructor for `boost::filesystem::path` is called. And then a destructor for a local variable. The fact that object can be constructed inside vector storage probably is already used by compilers for push_back case because I couldn't get any significant performance difference for just replacing push_back by emplace_back: https://quick-bench.com/q/lUMW7TZktQLH978fpYoQ5BjvzKM . But with `std::move` it's obviously better.

Reply:
I suppose that I need to modify all other places where existing string variables are passed (if there are such other places), right?

Reply:
Isn't path an boost::filesystem::path object and not a string?
https://theboostcpplibraries.com/boost.filesystem-paths

Reply:
Any object that requires allocation. It's not just a string, but for a sake of simplicity I used `std::string` (quick-bench.com doesn't support boost). `boost::filesystem::path` has move constructor and it owns strings so the same rule can be applied.

Reply:
> I couldn't get any significant performance difference for just replacing push_back by emplace_back

Even on your chart there is a 6% boost, and we do not use `std::move` widely on our codebase.

Also personaly I do not like the `std::move` syntax.
As for me, the 
```
results.recordMapping.emplace_back(id, newId);
```
is more readable than
```
results.recordMapping.push_back(std::make_pair(id, newId));
```
but the 
```
#include <utility>
results.recordMapping.emplace_back(std::move(id), std::move(newId));
```
does not make code to look better at all.

Reply:
For readability, could a wrapper function be created for the cases where std::move() needs to be used inside of emplace_back()

Reply:
In case of `push_back` there is a call of copy constructor for path, constructor for pair and then move constructor for pair that includes calling move constructor for path. For `emplace_back` it's copy constructor for path and constructor for pair. So the difference is move constructor for pair is not called and move constructor for path. That can explain these 6%. But copy constructor call for path is the most expensive operation here.

`std::move` is a standard tool to call move constructors with lvalue. There is no other way to do it rather than implementing it yourself. When you use it other developers understand what's going on. If you try to do some tricks expect questions.

Also in this case you can do:
```c++
std::time_t time = lastWriteTime(path, defaultTime);
if (time != defaultTime)
{
    contentFiles.emplace_back(time, dataPath / *entry);
    found = true;
    break;
}
```

`dataPath / *entry` expression now returns rvalue so move constructor is called by default and there is no need in keeping `path` variable.

https://quick-bench.com/q/3uH25AnXvfKyl_Roc9HoPEkiL0o

Reply:
For the context of:
```c++
results.recordMapping.emplace_back(std::move(id), std::move(newId));
```

```c++
for (const std::string& id : ) {
    std::string newId = ...;
    results.recordMapping.emplace_back(std::move(id), std::move(newId));
    reverseLookupMap.emplace(texture, newId);
}
```
Not sure did you mean it or something else. `std::move(id)` doesn't make any sense, const modifier will not go away, the result will be of type `const std::string&&` which is implicitly converted into `const std::string&` and copy constructor will be called. `std::move(newId)` will leave an empty string in `newId` which is not something you want in this case.

Reply:
> dataPath / *entry expression now returns rvalue so move constructor is called by default and there is no need in keeping path variable.

`path` variable is used in the `lastWriteTime` method, so your code does not compile.

Reply:
Using `std::move(path)`, and I do not see a point to modify other places. It can be done via follow-up PR, if needed.

> For the context of

It is just an example of syntax.